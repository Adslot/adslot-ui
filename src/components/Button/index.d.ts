import * as React from 'react';

export type ButtonColor = 'default' | 'primary' | 'secondary' | 'success' | 'danger' | 'warning' | 'info';

export type ButtonVariant = 'solid' | 'borderless' | 'inverse' | 'link';

export type ButtonSize = 'medium' | 'large';

export type ButtonIconPosition = 'left' | 'right';

export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  isLoading?: boolean;
  color?: ButtonColor;
  variant?: ButtonVariant;
  size?: ButtonSize;
  iconPosition?: ButtonIconPosition;
  /**
   * @deprecated
   * Please use the `color` prop instead.
   */
  theme?: string;
  /**
   * @deprecated
   * Please use `variant="inverse"` instead.
   */
  inverse?: boolean;
  round?: boolean;
  icon?: React.ReactNode;
  fullWidth?: boolean;
  className?: string;
  dts?: string;
  disabled?: boolean;
  children?: React.ReactNode;
}

declare const Button: React.FC<ButtonProps>;

export default Button;
