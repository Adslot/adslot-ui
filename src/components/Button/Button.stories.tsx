import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';

import Button from './index';

const meta = {
  title: 'Components/Button',
  component: Button,
  tags: ['autodocs'],
  argTypes: {
    theme: { control: false },
    inverse: { control: false },
    children: { control: 'text' },
    icon: {
      options: ['Add Icon', 'Filter Icon'],
      control: { type: 'select' },
      mapping: {
        'Add Icon': (
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" className="svg-icon" width="16" height="16">
            <path d="M8 2c-.3 0-.6.3-.6.6v4.8H2.6c-.3 0-.6.3-.6.6s.3.6.6.6h4.8v4.8c0 .3.3.6.6.6.2 0 .3-.1.4-.2.1-.1.2-.3.2-.4V8.6h4.8c.2 0 .3-.1.4-.2.1-.1.2-.2.2-.4 0-.3-.3-.6-.6-.6H8.6V2.6c0-.3-.3-.6-.6-.6z"></path>
          </svg>
        ),
        'Filter Icon': (
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40.2 40.2" className="svg-icon" width="16" height="16">
            <path d="M3.4 7.2V33c0 2.3 1.8 4.1 4.1 4.1h25.8c2.3 0 4.1-1.8 4.1-4.1V7.2c0-2.3-1.8-4.1-4.1-4.1H7.5c-2.3 0-4.1 1.8-4.1 4.1zm29.9-2.7c1.5 0 2.7 1.2 2.7 2.7V33c0 1.5-1.2 2.7-2.7 2.7H7.5c-1.5 0-2.7-1.2-2.7-2.7V7.2c0-1.5 1.2-2.7 2.7-2.7h25.8z"></path>
            <path d="M10.1 27.2h-.5c-.4 0-.7.3-.7.7 0 .4.3.7.7.7h.4c.3 1.7 1.8 2.9 3.6 2.9s3.2-1.3 3.6-2.9H31c.4 0 .7-.3.7-.7 0-.4-.3-.7-.7-.7H17.2c-.3-1.7-1.8-2.9-3.6-2.9-1.7 0-3.2 1.2-3.5 2.9zm5.7.7c0 1.2-1 2.2-2.2 2.2-1.2 0-2.2-1-2.2-2.2s1-2.2 2.2-2.2c1.3 0 2.2 1 2.2 2.2zm6.7-8.6H9.6c-.4 0-.7.3-.7.7 0 .4.3.7.7.7h12.8c.3 1.7 1.8 2.9 3.6 2.9s3.2-1.3 3.6-2.9H31c.4 0 .7-.3.7-.7 0-.4-.3-.7-.7-.7h-1.4c-.3-1.7-1.8-2.9-3.6-2.9-1.7-.1-3.2 1.2-3.5 2.9zm5.7.7c0 1.2-1 2.2-2.2 2.2s-2.2-1-2.2-2.2c0-1.2 1-2.2 2.2-2.2s2.2.9 2.2 2.2zm-14.6-8.4h-4c-.4 0-.7.3-.7.7 0 .4.3.7.7.7h4c.3 1.7 1.8 2.9 3.6 2.9s3.2-1.3 3.6-2.9H31c.4 0 .7-.3.7-.7 0-.4-.3-.7-.7-.7H20.7c-.3-1.7-1.8-2.9-3.6-2.9s-3.2 1.3-3.5 2.9zm5.8.7c0 1.2-1 2.2-2.2 2.2s-2.2-1-2.2-2.2 1-2.2 2.2-2.2 2.2 1 2.2 2.2z"></path>
          </svg>
        ),
      },
    },
    round: { if: { arg: 'icon' } },
  },
} satisfies Meta<typeof Button>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    variant: 'solid',
    children: 'Button',
    color: 'default',
  },
};

export const Solid: Story = {
  args: {
    variant: 'solid',
    children: 'Solid',
  },
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/ILhnLAeRObDeYARnBkHJeN/Solid-Buttons?type=design&node-id=0%3A1&mode=design&t=HMXgDOa3qkm4yVem-1',
    },
  },
  render: (args) => (
    <>
      <Button {...args} color="default" />
      <Button {...args} color="primary" />
      <Button {...args} color="secondary" />
      <Button {...args} color="danger" />
      <Button {...args} color="info" />
      <Button {...args} color="success" />
      <Button {...args} color="warning" />
    </>
  ),
};

export const Borderless: Story = {
  args: {
    variant: 'borderless',
    children: 'Borderless',
  },
  render: Solid.render,
};

export const Inverse: Story = {
  args: {
    variant: 'inverse',
    children: 'Inverse',
  },
  render: Solid.render,
};

/**
 * A link look-alike button. For actual href redirects, please use [Anchor](?path=/docs/components-anchor--docs) component
 */
export const Link: Story = {
  args: {
    variant: 'link',
    children: 'Link',
  },
  parameters: {
    controls: { exclude: ['color'] },
  },
};

export const FullWidth: Story = {
  args: {
    fullWidth: true,
    children: 'Full Width',
  },
  render: (args) => (
    <div style={{ display: 'flex', width: 600, gap: 24 }}>
      <Button {...args}>Secondary</Button>
      <Button {...args} color="primary">
        Primary
      </Button>
    </div>
  ),
};

export const IconButton: Story = {
  args: {
    'aria-label': 'add',
    icon: 'Add Icon',
  },
  argTypes: {
    'aria-label': { type: { required: true, name: 'string' }, description: 'Controls the label for the icon button.' },
  },
  render: Solid.render,
};

/**
 * Round buttons are only available when there is only icon
 */
export const IconButtonRound: Story = {
  args: {
    'aria-label': 'add',
    icon: 'Add Icon',
    round: true,
  },
  argTypes: {
    'aria-label': { type: { required: true, name: 'string' } },
  },
  parameters: {
    controls: { exclude: ['children'] },
  },
  render: Solid.render,
};

export const IconButtonWithLabel: Story = {
  args: {
    icon: 'Filter Icon',
    children: 'Filter',
  },
  parameters: {
    controls: { exclude: ['round'] },
  },
};
