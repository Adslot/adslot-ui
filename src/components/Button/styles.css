@import '../../styles/variable';

$font-size-button-default: 12px;
$font-size-button-large: 15px;
$border-radius-button-default: 2px;
$border-radius-button-large: 4px;
$line-height-button: 1.5;
$line-height-button-large: 1.467;
$gap-button-default: 6px;
$gap-button-large: 9px;
$size-button-default: 30px;
$size-button-large: 42px;
/* calc(($size-button-default - 2px - $font-size-button-default * $line-height-button) / 2) */
$padding-y-button-default: 5px;
$padding-x-button-default: 12px;
$padding-icon-button-default: 6px;
$padding-icon-button-large: 10px;

/* calc(($size-button-large - 2px - $font-size-button-large * $line-height-button-large) / 2) */
$padding-y-button-large: 9px;
$padding-x-button-large: 24px;
$focus-ring-color: $color-gray-darker;

.aui--button,
.aui--anchor,
.btn {
  display: inline-flex;
  grid-gap: $gap-button-default;
  align-items: center;
  justify-content: center;
  margin-bottom: 0;
  font-weight: 700;
  text-align: center;
  white-space: nowrap;
  text-decoration: none;
  min-height: $size-button-default;
  height: 0%;
  position: relative;
  vertical-align: middle;
  background-image: none;
  border: 1px solid transparent;
  padding: $padding-y-button-default $padding-x-button-default;
  font-size: $font-size-button-default;
  line-height: $line-height-button;
  border-radius: $border-radius-button-default;
  transition-timing-function: ease;
  transition-property: background-color, border-color, box-shadow, transform;
  transition-duration: 0.1s;
  margin-right: 5px;
  color: $color-gray-darker;
  background-color: $color-inverse;
  border-color: $color-gray-light;

  &:last-child {
    margin-right: 0;
  }

  &:focus,
  &:hover {
    text-decoration: none;
  }

  &:focus-visible {
    outline: none;
    border-color: $color-inverse;
    box-shadow: 0 0 0 2px $focus-ring-color;
  }

  &[class*='aui-'] {
    &:focus-visible {
      border-color: $color-inverse;
    }
  }

  &:hover {
    color: $color-gray-darker;
    background-color: $color-gray-lighter;
    border-color: #bababa;
  }

  &.aui-primary {
    color: $color-inverse;
    background-color: $color-primary;
    border-color: $color-primary;

    &:hover {
      color: $color-inverse;
      background-color: #005299;
      border-color: #005299;
    }

    &:active {
      color: $color-inverse;
      background-color: #004480;
      border-color: #004480;
    }
  }

  &.aui-secondary {
    color: $color-inverse;
    background-color: $color-secondary;
    border-color: $color-secondary;

    &:hover {
      color: $color-inverse;
      background-color: #008577;
      border-color: #008577;
    }

    &:active {
      color: $color-inverse;
      background-color: #006c60;
      border-color: #006c60;
    }
  }

  &.aui-success {
    color: $color-inverse;
    background-color: $color-positive;
    border-color: $color-positive;

    &:hover {
      color: $color-inverse;
      background-color: #449b44;
      border-color: #449b44;
    }

    &:active {
      color: $color-inverse;
      background-color: #3c893c;
      border-color: #3c893c;
    }
  }

  &.aui-info {
    color: $color-inverse;
    background-color: $color-info;
    border-color: $color-info;

    &:hover {
      color: $color-inverse;
      background-color: #3195b2;
      border-color: #3195b2;
    }

    &:active {
      color: $color-inverse;
      background-color: #2b849e;
      border-color: #2b849e;
    }
  }

  &.aui-warning {
    color: $color-inverse;
    background-color: $color-warning;
    border-color: $color-warning;

    &:hover {
      color: $color-inverse;
      background-color: #f39106;
      border-color: #f39106;
    }

    &:active {
      color: $color-inverse;
      background-color: #da8205;
      border-color: #da8205;
    }
  }

  &.aui-danger {
    color: $color-inverse;
    background-color: $color-negative;
    border-color: $color-negative;

    &:hover {
      color: $color-inverse;
      background-color: #c72f29;
      border-color: #c72f29;
    }

    &:active {
      color: $color-inverse;
      background-color: #b22a24;
      border-color: #b22a24;
    }
  }

  &.aui-inverse {
    background-color: $color-inverse;
    border-color: $color-gray-light;

    &:focus-visible {
      border-color: $color-inverse;
      box-shadow: 0 0 0 2px $focus-ring-color;
    }

    &:hover {
      background-color: $color-gray-lighter;
      border-color: #bababa;
    }

    &:active {
      color: $color-inverse;
      background-color: #dbdbdb;
      border-color: #bababa;
    }
  }

  &.aui-borderless {
    background-color: transparent;
    border-color: transparent;

    &:focus-visible {
      border-color: $color-inverse;
      box-shadow: 0 0 0 2px $focus-ring-color;
    }

    &:hover {
      background-color: $color-inverse;
      border-color: $color-gray-light;
    }

    &:active {
      background-color: $color-gray-lighter;
      border-color: #adadad;
    }
  }

  &.aui-borderless,
  &.aui-inverse {
    &.aui-primary {
      color: $color-primary;

      &:hover {
        color: $color-primary;
      }
    }

    &.aui-default {
      color: $color-gray-darker;

      &:hover {
        color: $color-gray-darker;
      }
    }

    &.aui-secondary {
      color: $color-secondary;

      &:hover {
        color: $color-secondary;
      }
    }

    &.aui-success {
      color: $color-positive;

      &:hover {
        color: $color-positive;
      }
    }

    &.aui-info {
      color: $color-info;

      &:hover {
        color: $color-info;
      }
    }

    &.aui-warning {
      color: $color-warning;

      &:hover {
        color: $color-warning;
      }
    }

    &.aui-danger {
      color: $color-negative;

      &:hover {
        color: $color-negative;
      }
    }
  }

  /* Link */
  &.aui-link {
    all: unset;
    border-radius: $border-radius-button-default;
    color: $color-primary;
    text-decoration: none;
    cursor: pointer;

    &:hover {
      color: #004c8f;
    }

    &:focus-visible {
      box-shadow: 0 0 0 2px $color-gray-darker;
    }
  }

  /* Large */
  &.aui-large {
    line-height: $line-height-button-large;
    min-height: $size-button-large;
    padding: $padding-y-button-large $padding-x-button-large;
    font-size: $font-size-button-large;
    grid-gap: $gap-button-large;

    &.aui-icon {
      padding: $padding-icon-button-large;
      height: $size-button-large;
      width: $size-button-large;
    }
  }

  /* Round */
  &.aui-round {
    border-radius: 999px;
  }

  /* Full Width */
  &.aui-full-width {
    width: 100%;
    margin-right: 0;
  }

  /* Icon (no label/children) */
  &.aui-icon {
    padding: $padding-icon-button-default;
    height: $size-button-default;
    width: $size-button-default;

    &.aui-round {
      & .spinner-component {
        & .spinner {
          &.spinner-small {
            height: $size-button-default;
            width: $size-button-default;
          }

          &.spinner-medium {
            border-width: 3px;
            height: $size-button-large;
            width: $size-button-large;
          }
        }
      }
    }
  }

  /* Icon container */
  & .aui-icon-container {
    display: inline-flex;

    & svg {
      fill: currentColor;
    }
  }

  /* Disabled */
  &[class*='aui-'].disabled,
  &[class*='aui-'][disabled] {
    background-color: $color-gray-lighter;
    border-color: $color-gray-lighter;
    color: $color-gray-dark;
    opacity: 0.65;
    box-shadow: none;
    cursor: not-allowed;

    &.aui-link {
      background-color: transparent;
    }

    &:hover,
    &:focus {
      background-color: $color-gray-lighter;
      border-color: $color-gray-lighter;
      color: $color-gray-dark;
      opacity: 0.65;
      box-shadow: none;
      cursor: not-allowed;

      &.aui-link {
        background-color: transparent;
      }
    }

    &:focus {
      box-shadow: none;
    }
  }

  /* Loading */
  &:not(.aui-round) {
    & .aui-children-container,
    & .aui-icon-container {
      /* preserves width and height of button */
      &.is-loading {
        visibility: hidden;
      }

      & > .aui--svg-symbol-component {
        display: block;
        margin: auto;
      }
    }
  }

  & .spinner-container {
    position: absolute;

    & .spinner-component {
      display: flex;
      align-items: center;
      justify-content: center;
    }
  }

  & > .flexible-wrapper-inline {
    & > .flexible-spacer-component {
      min-width: $padding-base-horizontal;
    }
  }
}
