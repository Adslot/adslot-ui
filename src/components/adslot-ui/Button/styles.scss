@import '~styles/variable';

$default-background-color: $color-white;
$default-text-color: $color-gray-darker;
$default-border-color: $color-gray-lighter;

// Mixin Function
@mixin bubble-keyframes($style, $bubble-color) {
  @keyframes bubble-#{$style} {
    0% {
      width: 0px;
      padding-top: 0px;
      background: rgba($bubble-color, 0.5);
    }

    80% {
      width: 100%;
      padding-top: 100%;
      background: rgba($bubble-color, 0.1);
    }

    100% {
      width: 100%;
      padding-top: 100%;
      background: transparent;
    }
  }
}

@mixin focusWithBubble($style, $bubble-color) {
  @include bubble-keyframes($style, $bubble-color);

  &:hover,
  &:focus {
    &:before {
      content: '';
      position: absolute;
      border-radius: 50%;
      animation: bubble-#{$style} 0.7s;
    }
  }
}

@mixin btn-style($style, $background-color) {
  &[data-is-loading="false"].btn-is-#{$style}:not(.aui-component--is-disabled) {
    .btn-label {
      background-color: $background-color;
      color: $color-white;
      border: $background-color 1px solid;

      &:active {
        background-color: darken($background-color, 20%);
      }

      @include focusWithBubble('light', $color-white);
    }
  }

  &[data-is-loading="false"].btn-is-inverse-#{$style}:not(.aui-component--is-disabled) {
    .btn-label {
      background-color: $default-background-color;
      color: $background-color;
      border: $default-border-color 1px solid;

      &:active {
        background-color: $default-border-color;
      }

      @include focusWithBubble('dark', $default-border-color);
    }
  }
}

// Main style
.aui-component--btn {
  display: inline-block;
  position: relative;
  font-size: $font-size-body;
  font-weight: $font-weight-bold;
  line-height: 1;
  padding: 0;
  border: none;
  outline: none;
  overflow: hidden;

  .btn-label {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 30px;
    min-width: 102px;
    padding-left: 12px;
    padding-right: 12px;
    border-radius: 0.3rem;
  }

  @include btn-style('primary', $color-blue);

  @include btn-style('alert', $color-red);

  @include btn-style('success', $color-green);

  &.aui-component--is-disabled {
    .btn-label {
      background-color: $color-gray-lighter;
      color: $color-gray;
      border: $color-gray-lighter 1px solid;

      &:hover {
        cursor: not-allowed;
      }
    }
  }

  &[data-is-loading="false"]:not(.aui-component--is-disabled) {
    .btn-label {
      background-color: $default-background-color;
      color: $default-text-color;
      border: $default-border-color 1px solid;

      &:active {
        background-color: darken($default-background-color, 20%);
      }

      @include focusWithBubble('dark', $default-border-color);
    }
  }

  &[data-is-loading="true"][data-is-loading="true"] {
    .btn-label {
      background-color: $color-gray-lighter;
      border-color: $color-gray;
      transition: background-color, border-color;
      transition-timing-function: ease-in-out;
      transition-duration: 0.5s;
    }
  }

  &.aui-component--btn-sm {
    .btn-label {
      height: 24px;
      min-width: 72px;
    }
  }
}
