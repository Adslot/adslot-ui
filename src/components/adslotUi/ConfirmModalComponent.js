import Button from 'react-bootstrap/lib/Button';
import Modal from 'react-bootstrap/lib/Modal';
import React, { PropTypes } from 'react';

const ConfirmModalComponent = ({
  buttonCancelLabel,
  buttonConfirmLabel,
  modalApply,
  modalClose,
  modalDescription,
  modalTitle,
  show,
}) => {
  const cancelAction = () => { modalClose(); };

  const applyAction = () => {
    modalApply();
    modalClose();
  };

  return (
    <Modal className="confirm-modal-component" show={show} bsSize="small" keyboard={false}>
      {modalTitle ? <Modal.Header><Modal.Title>{modalTitle}</Modal.Title></Modal.Header> : null}
      <Modal.Body>
        <p>{modalDescription}</p>
      </Modal.Body>
      <Modal.Footer>
        {modalClose ? <Button className="btn-inverse" onClick={cancelAction} data-test-selector="confirm-modal-cancel">
          {buttonCancelLabel}</Button> : null}
        <Button bsStyle="primary" onClick={applyAction} data-test-selector="confirm-modal-confirm">
          {buttonConfirmLabel}</Button>
      </Modal.Footer>
    </Modal>
  );
};

ConfirmModalComponent.displayName = 'AdslotUiConfirmModalComponent';

ConfirmModalComponent.propTypes = {
  buttonCancelLabel: PropTypes.string.isRequired,
  buttonConfirmLabel: PropTypes.string.isRequired,
  modalApply: PropTypes.func.isRequired,
  modalClose: PropTypes.func,
  modalDescription: PropTypes.string.isRequired,
  modalTitle: PropTypes.string,
  show: PropTypes.bool.isRequired,
};

ConfirmModalComponent.defaultProps = {
  buttonCancelLabel: 'Cancel',
  buttonConfirmLabel: 'Confirm',
  modalApply: () => { throw new Error('AdslotUi ConfirmModal needs a modalApply handler'); },

  modalDescription: 'Are you sure?',
  show: false,
};

export default ConfirmModalComponent;
