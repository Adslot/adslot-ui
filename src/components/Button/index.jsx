/* eslint-disable react/prop-types */
// disable proptypes check because it doesn't take into consideration extended types
import _ from 'lodash';
import classNames from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';
import BootstrapButton from 'react-bootstrap/lib/Button';
import Spinner from '../Spinner';
import { expandDts } from '../../lib/utils';
import './styles.scss';

const adslotButtonPropTypes = {
  /**
   * PropTypes.oneOf(['small', 'large'])
   */
  size: PropTypes.oneOf(['small', 'large']),
  inverse: PropTypes.bool,
  dts: PropTypes.string,
  isLoading: PropTypes.bool,
  /**
   * PropTypes.oneOf(['primary', 'success', 'info', 'warning', 'danger', 'link'])
   */
  bsStyle: PropTypes.oneOf(['primary', 'success', 'info', 'warning', 'danger', 'link']),
};

const Button = props => {
  const { bsStyle, inverse, size, children, dts, className, isLoading, disabled } = props;
  const baseClass = 'aui--button';
  const classes = classNames(baseClass, className, {
    'btn-inverse': inverse && !/btn-inverse/.test(className),
    'btn-large': size === 'large',
    'aui--btn-default':
      (!bsStyle || bsStyle === 'default') &&
      (!className ||
        ['btn-default', 'btn-inverse', 'btn-default btn-inverse', 'btn-inverse btn-default'].includes(className)),
  });

  const renderSpinner = () =>
    isLoading ? (
      <div className="spinner-container">
        <Spinner size={_.includes(['lg', 'large'], props.bsSize) ? 'medium' : 'small'} />
      </div>
    ) : null;

  return (
    <BootstrapButton
      bsStyle={bsStyle}
      {..._.omit(props, _.keys(adslotButtonPropTypes))}
      disabled={isLoading || disabled}
      className={classes}
      {...expandDts(dts)}
    >
      {renderSpinner()}
      <div className={isLoading ? 'aui--button-children-container' : null}>{children}</div>
    </BootstrapButton>
  );
};

Button.propTypes = { ...adslotButtonPropTypes, ...BootstrapButton.propTypes };

Button.defaultProps = {
  inverse: false,
  isLoading: false,
  size: 'small',
};

export default Button;
