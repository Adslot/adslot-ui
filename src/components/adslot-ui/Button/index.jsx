import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { Spinner } from 'alexandria';
import './styles.scss';

const classPrefix = 'aui-component';

class Button extends React.PureComponent {
  render() {
    const { disabled, small, isLoading, inverse, children, style, onClick } = this.props;
    const isStyledBtn = style !== 'default';
    const className = classnames([
      `${classPrefix}--btn`,
      {
        [`${classPrefix}--btn-sm`]: small,
        [`${classPrefix}--is-disabled`]: disabled,
        [`btn-is-${style}`]: isStyledBtn && !inverse,
        [`btn-is-inverse-${style}`]: isStyledBtn && inverse,
      },
    ]);

    const buttonProps = {
      className,
      disabled,
      'data-is-loading': isLoading,
      ...(disabled || isLoading ? {} : { onClick }),
    };
    return (
      <button {...buttonProps}>
        <div className="btn-label">{isLoading ? <Spinner size="small" /> : children}</div>
      </button>
    );
  }
}

Button.defaultProps = {
  small: false,
  isLoading: false,
  style: 'default',
};

Button.propTypes = {
  disabled: PropTypes.bool,
  small: PropTypes.bool,
  isLoading: PropTypes.bool,
  inverse: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
  style: PropTypes.oneOf(['primary', 'default', 'warning', 'alert', 'success']),
  onClick: PropTypes.func,
};

export default Button;
