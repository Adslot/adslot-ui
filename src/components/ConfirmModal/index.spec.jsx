import _ from 'lodash';
import Button from 'react-bootstrap/lib/Button';
import ConfirmModalComponent from 'components/ConfirmModal';
import Modal from 'react-bootstrap/lib/Modal';
import React from 'react';
import { shallow } from 'enzyme';

describe('ConfirmModalComponent', () => {
  it('should render with defaults', () => {
    const component = shallow(<ConfirmModalComponent />);
    expect(component.prop('className')).to.equal('confirm-modal-component');
    expect(component.type()).to.equal(Modal);
    expect(component.prop('show')).to.equal(false);
    expect(component.prop('bsSize')).to.equal('small');
    expect(component.prop('keyboard')).to.equal(false);

    const modalHeaderElement = component.find(Modal.Header);
    expect(modalHeaderElement).to.have.length(0);

    const modalBodyElement = component.find(Modal.Body);
    const modalDescriptionElement = modalBodyElement.children();
    const modalDescriptionText = modalDescriptionElement.text();
    expect(modalDescriptionText).to.equal('Are you sure?');

    const modalFooterElement = component.find(Modal.Footer);
    const applyButtonElement = modalFooterElement.find(Button).first();
    expect(applyButtonElement.prop('bsStyle')).to.equal('primary');
    expect(applyButtonElement.prop('onClick')).to.be.a('function');
    expect(applyButtonElement.children().text()).to.equal('Confirm');
  });

  it('should render with props', () => {
    const props = {
      buttonCancelLabel: 'Close',
      buttonConfirmLabel: 'OK',
      modalClose: _.noop,
      modalDescription: 'If sure, please click confirm.',
      modalTitle: 'Please Confirm',
    };
    const component = shallow(<ConfirmModalComponent {...props} />);
    expect(component.prop('className')).to.equal('confirm-modal-component');

    const modalHeaderElement = component.find(Modal.Header);
    const modalTitleElement = modalHeaderElement.find(Modal.Title);
    const modalTitleText = modalTitleElement.children().text();
    expect(modalTitleText).to.equal('Please Confirm');

    const modalBodyElement = component.find(Modal.Body);
    const modalDescriptionElement = modalBodyElement.children();
    const modalDescriptionText = modalDescriptionElement.text();
    expect(modalDescriptionText).to.equal('If sure, please click confirm.');

    const modalFooterElement = component.find(Modal.Footer);
    const cancelButtonElement = modalFooterElement.find(Button).first();
    expect(cancelButtonElement.prop('className')).to.equal('btn-inverse');
    expect(cancelButtonElement.prop('onClick')).to.be.a('function');
    expect(cancelButtonElement.children().text()).to.equal('Close');

    const applyButtonElement = modalFooterElement.find(Button).last();
    expect(applyButtonElement.prop('bsStyle')).to.equal('primary');
    expect(applyButtonElement.prop('onClick')).to.be.a('function');
    expect(applyButtonElement.children().text()).to.equal('OK');
  });

  it('should show modal when `show` is true', () => {
    const component = shallow(<ConfirmModalComponent show />);
    expect(component.prop('show')).to.equal(true);
  });

  it('should hide modal when `show` is false', () => {
    const component = shallow(<ConfirmModalComponent show={false} />);
    expect(component.prop('show')).to.equal(false);
  });

  it('should call `modalApply` and `modalClose` when we click Apply', () => {
    let applyCalls = 0;
    let closeCalls = 0;
    const applyMock = () => {
      applyCalls++;
    };

    const closeMock = () => {
      closeCalls++;
    };

    const component = shallow(<ConfirmModalComponent modalApply={applyMock} modalClose={closeMock} />);
    const modalFooterElement = component.find(Modal.Footer);
    const applyButtonElement = modalFooterElement.find(Button).last();
    applyButtonElement.simulate('click');
    expect(applyCalls).to.equal(1);
    expect(closeCalls).to.equal(1);
  });

  it('should throw when we click Apply without a handler', () => {
    const component = shallow(<ConfirmModalComponent />);
    const modalFooterElement = component.find(Modal.Footer);
    const applyButtonElement = modalFooterElement.find(Button).first();
    expect(applyButtonElement.prop('onClick')).to.throw('AdslotUi ConfirmModal needs a modalApply handler');
  });

  it('should call `modalClose` when we click Cancel', () => {
    let closeCalls = 0;
    const closeMock = () => {
      closeCalls += 1;
    };

    const component = shallow(<ConfirmModalComponent modalClose={closeMock} />);
    const modalFooterElement = component.find(Modal.Footer);
    const cancelButtonElement = modalFooterElement.find(Button).first();
    cancelButtonElement.simulate('click');
    expect(closeCalls).to.equal(1);
  });
});
