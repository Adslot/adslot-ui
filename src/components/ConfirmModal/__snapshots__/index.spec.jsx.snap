// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmModal /> should show modal when \`show\` is true 1`] = `
<div
  class="confirm-modal-component fade in modal"
  data-testid="confirm-modal-wrapper"
  role="dialog"
  style="display: block; padding-right: 0px;"
  tabindex="-1"
>
  <div
    class="modal-sm modal-dialog"
  >
    <div
      class="modal-content"
      role="document"
    >
      <div
        class="modal-body"
        data-testid="confirm-modal-body"
      >
        <p>
          Are you sure?
        </p>
      </div>
      <div
        class="modal-footer"
        data-testid="confirm-modal-footer"
      >
        <button
          class="btn btn-primary"
          data-test-selector="confirm-modal-confirm"
          data-testid="confirm-modal-confirm"
          type="button"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
